posts/A-roadmap-to-TDD-adoption.md

```diff
@@ -0,0 +1,53 @@
+---
+authors:
+- Ben Scott
+tags:
+- Agile
+- XP
+date: 
+title: "A roadmap to TDD adoption"
+image: https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2018/08/TDD_Graphic.png
+---
+
+You've heard all about how Test Driven Development (TDD) will solve all your problems on a software project. Your defects go away, your time to market decreases, customer satisfaction goes through the roof. Everything is supposed to be awesome if your team just did TDD. However every time you try to adopt it, the adoption fails. Why?
+
+# Test Driven Development
+
+Let's just start with what is TDD. This practice states that for any code that you write you first start writing the unit test for it. This test should fail, then you write the code that will satisfy the test, if needed you refactor and finally you write the next failing test and start this cycle again.
+
+The main reason for adoption is your IT manager or tech lead has read a blog or attended a talk on how TDD solves these main problems:
+
+* Too many defects in production
+* Long QA cycles
+* New features break old features
+* Devops adoption
+
+# Traditional approach
+
+The typical adoption pattern is to start with some classroom training on how to write unit tests, and on how to mock. You follow up with some coding katas where the team has to write a common software problem using TDD (fizz-buzz, prime numbers, bowling score calculator, etc...). Once you've got a handle on those exercises you're supposed to take that back to your project and be able to develop 'real' code using TDD. This almost always fails. First, production code is much more complicated than exercises, than code that isn't written with testing in mind is ridiculously hard to do TDD with, and finally TDD isn't actually about testing and most teams focus on writing tests for test coverage and forget the most important part of the cycle. Refactoring.
```

Comment:
I think "...the team has to write a common software problem using TDD..." should be something like "...the team has to solve a common software problem by writing code using TDD..."